src/diagnostics_component/utils/file_upload_utils/src/blob_storage_helper.cpp

```diff
@@ -0,0 +1,105 @@
+ * @param blobInfo information related to the blob storage account
+ * @param maxConcurrency the maximum number of threads that can be working at one time for the client connection
+ */
+AzureBlobStorageHelper::AzureBlobStorageHelper(const BlobStorageInfo& blobInfo, const unsigned int maxConcurrency)
+{
+    if (blobInfo.storageSasCredential == nullptr)
+    {
+        throw std::invalid_argument("Container name invalid");
+    }
+
+    // NOLINTNEXTLINE (modernize-make-unique): There is no std::make_unique in c++11
+    client = std::make_unique<Azure::Storage::Blobs::BlobContainerClient>(STRING_c_str(blobInfo.storageSasCredential));
+}
+
+/**
+ * @brief strips the file name from the @p filePath
+ * @param fileName a properly formed name for the file who's path is being constructed- must include extension if the file has one
+ * @param directoryPath a properly formed path to the directory where the file exists
+ * @returns the filePath, or ""
+ */
+std::string
+AzureBlobStorageHelper::CreatePathFromFileAndDirectory(const std::string& fileName, const std::string& directoryPath)
+{
+    if (fileName.empty() || directoryPath.empty())
+    {
+        throw std::runtime_error("Invalid Arguments to CreatePathFromFileAndDirectory");
+    }
+
+    _Bool needsSeperator = false;
+    if (directoryPath.find_last_of('/') != directoryPath.length() - 1)
+    {
+        needsSeperator = true;
+    }
+
+    if (needsSeperator)
+    {
+        return directoryPath + "/" + fileName;
+    }
+    return directoryPath + fileName;
+}
+
+/**
+ * @brief Uploads all the files listed in @p files using the storage account associated with this object
+ * @param fileNames vector of file names to upload
+ * @param directoryPath path to the directory where @p fileNames can be found
+ * @param virtualDirectory a properly formatted virtual directory (ending in '/') to be used when uploading the files
+ * @returns true on success; false on any failure
+ */
+bool AzureBlobStorageHelper::UploadFilesToContainer(
+    VECTOR_HANDLE fileNames, const std::string& directoryPath, const std::string& virtualDirectory)
+{
+    if (VECTOR_size(fileNames) == 0)
+    {
+        throw std::runtime_error("Invalid Arguments to CreatePathFromFileAndDirectory");
+    }
+
+    std::string virtualDirectoryPath = virtualDirectory;
+    if (!virtualDirectoryPath.empty() && virtualDirectoryPath[virtualDirectoryPath.length() - 1] != '/')
+    {
+        virtualDirectoryPath += "/";
+    }
+
+    size_t fileNameSize = VECTOR_size(fileNames);
+    for (unsigned int i = 0; i < fileNameSize; ++i)
+    {
+        auto fileNameHandle = static_cast<const STRING_HANDLE*>(VECTOR_element(fileNames, i));
+        const char* fileName = STRING_c_str(*fileNameHandle);
+
+        std::string filePath;
+
+        try
+        {
+            filePath = CreatePathFromFileAndDirectory(fileName, directoryPath);
+        }
+        catch (std::exception& e)
```

Comment:
Windows build is failing because of unused parameter.  If taking suggestion in next comment, then it will be used when tracing e.what()

```
D:\a\1\s\src\diagnostics_component\utils\file_upload_utils\src\blob_storage_helper.cpp(90,32): 
warning C4101: 'e': unreferenced local variable 
[D:\a\1\s\out\Debug\src\diagnostics_component\utils\file_upload_utils\file_upload_utility.vcxproj]
```